ports/nrf/common-hal/busio/I2C.c

```diff
@@ -94,11 +94,19 @@ static uint8_t twi_error_to_mp(const nrfx_err_t err) {
     return 0;
 }
 
-void common_hal_busio_i2c_construct(busio_i2c_obj_t *self, const mcu_pin_obj_t *scl, const mcu_pin_obj_t *sda, uint32_t frequency, uint32_t timeout) {
+void common_hal_busio_i2c_construct(busio_i2c_obj_t *self, const mcu_pin_obj_t *scl, const mcu_pin_obj_t *sda, bool internal_pullup, uint32_t frequency, uint32_t timeout) {
     if (scl->number == sda->number) {
         mp_raise_ValueError(translate("Invalid pins"));
     }
 
+    // Set pulls on I2C pins up if setpullup is 1 - this supports boards with internal pullups
+    if (internal_pullup){
+        nrf_gpio_pin_pull_t hal_pull = NRF_GPIO_PIN_PULLUP;
+
+        nrf_gpio_cfg_input(scl->number, hal_pull);
+        nrf_gpio_cfg_input(sda->number, hal_pull);
+    }
+
```

Comment:
Move this down to line 124, and add an `else` for lines 125-136; do the check for external pullups only if `internal_pullup` is `false`.

Take away the `#if CIRCUITPY_REQUIRE_I2C_PULLUPS`; it is superseded by the passed-in boolean.

You don't need the `hal_pull` variable: you can use the constant:
```suggestion
    // Set pulls on I2C pins up if setpullup is 1 - this supports boards with internal pullups
    if (internal_pullup){
        nrf_gpio_cfg_input(scl->number, NRF_GPIO_PIN_PULLUP);
        nrf_gpio_cfg_input(sda->number, NRF_GPIO_PIN_PULLUP);
    }

```

Reply:
If move this up to line 124 and remove the `#if CIRCUITPY_REQUIRE_I2C_PULLUPS`, won't the pin pulls be set down after I set them up by the code following the `else` you suggest?

I think you mean I should put all of the test code into the `else`:
```
  // Set pulls up if internal_pullup is true
    if (internal_pullup){
        nrf_gpio_cfg_input(scl->number, NRF_GPIO_PIN_PULLUP);
        nrf_gpio_cfg_input(sda->number, NRF_GPIO_PIN_PULLUP);
    }
    else (!nrf_gpio_pin_read(sda->number) || !nrf_gpio_pin_read(scl->number)) {
         // Otherwise, test that the pins are in a high state. (Hopefully indicating they are pulled up.)
        nrf_gpio_cfg_input(scl->number, NRF_GPIO_PIN_PULLDOWN);
        nrf_gpio_cfg_input(sda->number, NRF_GPIO_PIN_PULLDOWN);

        common_hal_mcu_delay_us(10);

        nrf_gpio_cfg_input(scl->number, NRF_GPIO_PIN_NOPULL);
        nrf_gpio_cfg_input(sda->number, NRF_GPIO_PIN_NOPULL);

        // We must pull up within 3us to achieve 400khz.
        common_hal_mcu_delay_us(3);
        
        reset_pin_number(sda->number);
        reset_pin_number(scl->number);
        mp_raise_RuntimeError(translate("No pull up found on SDA or SCL; check your wiring"));
    }
```

Not second guessing you, just trying to understand.